test/support/component_helpers.rb

```diff
@@ -7,54 +7,38 @@ def assert_has_component_title(title, context = nil)
     end
   end
 
-  def assert_has_button_component(text, attrs = {})
-    all(shared_component_selector('button')).each do |button|
-      data = JSON.parse(button.text).symbolize_keys
-      if attrs[:href] && data[:href]
-        next unless text == data.delete(:text) && attrs[:href] == data[:href]
-      else
-        next unless text == data.delete(:text)
-      end
-      data.delete(:data_attributes) if data[:data_attributes].blank?
-      return assert_equal attrs, data
-    end
+  def assert_has_button_component(text)
+    assert page.has_css?("button", text: text)
+  end
 
-    fail_button_not_found(text)
+  def assert_has_button_component_as_link(text, attrs = {})
+    assert page.has_css?(process_button_attributes(attrs), text: text)
   end
 
   def refute_has_button_component(text, attrs = {})
-    all(shared_component_selector('button')).each do |button|
-      data = JSON.parse(button.text).symbolize_keys
-      if attrs[:href] && data[:href]
-        next unless text == data.delete(:text) && attrs[:href] == data[:href]
-      else
-        next unless text == data.delete(:text)
-      end
-      data.delete(:data_attributes) if data[:data_attributes].blank?
-      return refute_equal attrs, data
-    end
+    assert page.has_no_css?(process_button_attributes(attrs), text: text)
   end
 
   def click_button_component(text)
-    selector = shared_component_selector("button")
-
-    all(selector).each do |button|
-      data = JSON.parse(button.text).symbolize_keys
+    if page.has_css?("a.gem-c-button")
+      click_link text
+    else
+      click_button text
+    end
+  end
 
-      next if data[:text] != text
+  def process_button_attributes(attrs)
```

Comment:
This couples the tests really tightly to the CSS/HTML of the component. Would we be able to test whatever we want to test at a higher level? For example, by clicking on the button to check that something happens?

Reply:
I was hoping to achieve this but didn't know how. I figured that at least the only instances of this were in one place so could easily be changed.

There is this one test that is very specific about this, which is partly why I did this...

https://github.com/alphagov/frontend/blob/master/test/integration/licence_test.rb#L1195-L1212

Reply:
@tijmenb this thread has been marked as outdated even though it isn't, I think it's the only issue blocking this PR. Shall we catch up offline?